.github/actions/docker/build-api/action.yml

```diff
@@ -0,0 +1,81 @@
+name: Build API Docker Image
+
+description: 'Builds a docker image for the API'
+
+inputs:
+  tag:
+    description: 'The tag to use for the image'
+    required: false
+  push:
+    description: 'Push the image to the registry'
+    required: false
+    default: 'false'
+  github_token:
+    description: 'The token to use for logging into ghcr.io'
+    required: true
+
+outputs:
+  image:
+    description: 'The image that was built'
+    value: ${{ steps.build-image.outputs.IMAGE }}
+
+runs:
+  using: composite
+  steps:
+    - name: ⛏️ build api
+      shell: bash
+      run: pnpm build:api
+
+    - name: 🚢 Setup Depot
+      uses: depot/setup-action@v1
+      with:
+        oidc: true
+    - name: Login To Registry
+      shell: bash
+      env:
+        GH_ACTOR: ${{ github.actor }}
+        GH_PASSWORD: ${{ inputs.github_token }}
+      run: |
+        echo $GH_PASSWORD | docker login ghcr.io -u $GH_ACTOR --password-stdin
+
+    - name: Build, tag, and push image to Amazon ECR
```

Comment:
should we do the push part here optional as we do on ghcr push (on the next step)?

Reply:
Good catch! Exactly the problem